overlord/devicestate/handlers_install.go

```diff
@@ -913,17 +925,75 @@ func (m *DeviceManager) loadAndMountSystemLabelSnaps(systemLabel string) (*syste
 
 	seedSnaps := systemAndSnaps.SeedSnapsByType
 
-	for _, seedSn := range []*seed.Snap{seedSnaps[snap.TypeGadget], seedSnaps[snap.TypeKernel]} {
-		mntPt, unmountSnap, err := mountSeedSnap(seedSn)
+	// We might need mount for kernel and gadget only
+	for _, typ := range essentialTypes {
+		if typ != snap.TypeGadget && typ != snap.TypeKernel {
+			continue
+		}
+
+		seedSn := seedSnaps[typ]
+		mntPt, unmountSnap, err := mountSeedContainer(seedSn.Path, string(seedSn.EssentialType))
 		if err != nil {
 			unmount()
-			return nil, nil, nil, err
+			return nil, nil, nil, nil, err
 		}
 		unmountFuncs = append(unmountFuncs, unmountSnap)
 		mntPtForType[seedSn.EssentialType] = mntPt
 	}
 
-	return systemAndSnaps, mntPtForType, unmount, nil
+	mntPtForComp := make(map[string]string)
+	for _, seedComp := range systemAndSnaps.CompsByType[snap.TypeKernel] {
+		if seedComp.CompInfo.Type != snap.KernelModulesComponent {
+			continue
+		}
+
+		mntPt, unmountComp, err := mountSeedContainer(seedComp.CompSeed.Path,
+			seedComp.CompInfo.FullName())
+		if err != nil {
+			unmount()
+			return nil, nil, nil, nil, err
+		}
+		unmountFuncs = append(unmountFuncs, unmountComp)
+		mntPtForComp[seedComp.CompInfo.FullName()] = mntPt
+	}
+
+	return systemAndSnaps, mntPtForType, mntPtForComp, unmount, nil
+}
+
+func kModsInfo(systemAndSnaps *systemAndEssentialSnaps, kernMntPoint string, mntPtForComps map[string]string, isCore bool) (*install.KernelSnapInfo, []boot.BootableKModsComponents) {
+	kernInfo := systemAndSnaps.InfosByType[snap.TypeKernel]
+
+	// Find out kernel-modules components in the seed
+	compSeedInfos := systemAndSnaps.CompsByType[snap.TypeKernel]
+	bootKMods := make([]boot.BootableKModsComponents, 0, len(compSeedInfos))
```

Comment:
wouldn't make sense to add a field for this to KernelSnapInfo ? want we need it for when we deal with installing from snap-bootstrap or the API?

Reply:
thanks, fair enough, please double check again then, maybe instead we need a wrapper struct that carries both?

Reply:
Exactly, I was thinking yesterday something like that